esphome/components/pulse_meter/pulse_meter_sensor.cpp

```diff
@@ -58,8 +55,9 @@ void PulseMeterSensor::loop() {
     }
     this->last_pin_val_ = current;
 
-    // Swap out set and get to get the latest state from the ISR
-    std::swap(this->set_, this->get_);
+    // Get the latest state from the ISR and reset the count in the ISR
+    std::memcpy((void *) this->get_, (void *) this->set_, sizeof(State));
```

Comment:
No need for the memcpy here, it's probably clearer to just use assignment
```suggestion
    *this->get_ = *this->set_;
```

Reply:
I completely agree with you and it was my first choice, but it is not possible becase these are volatile. The cast to void in memcpy is needed for the same reason.

Reply:
Making them non volatile members rather than pointers is for the best then. The variables themselves aren't volatile anyway just the pointers to them

Reply:
But because the pointers are volatile and we always use pointers to access the members, the compiler considers the members to be volatile as well. At least that is my understanding. But I want to believe the interrupt lock will be enough.